src/Workspaces/Core/Portable/Editing/ImportAdder.cs

```diff
@@ -15,82 +15,114 @@ public static class ImportAdder
         /// <summary>
         /// Adds namespace imports / using directives for namespace references found in the document.
         /// </summary>
-        public static async Task<Document> AddImportsAsync(Document document, OptionSet options = null, CancellationToken cancellationToken = default)
+        public static Task<Document> AddImportsAsync(Document document, OptionSet options = null, CancellationToken cancellationToken = default)
         {
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            return await AddImportsAsync(document, root.FullSpan, options, cancellationToken).ConfigureAwait(false);
+            return AddImportsFromSyntaxesAsync(document, false, options, cancellationToken);
         }
 
         /// <summary>
         /// Adds namespace imports / using directives for namespace references found in the document within the span specified.
         /// </summary>
         public static Task<Document> AddImportsAsync(Document document, TextSpan span, OptionSet options = null, CancellationToken cancellationToken = default)
         {
-            return AddImportsAsync(document, new[] { span }, options, cancellationToken);
+            return AddImportsFromSyntaxesAsync(document, new[] { span }, false, options, cancellationToken);
         }
 
         /// <summary>
         /// Adds namespace imports / using directives for namespace references found in the document within the sub-trees annotated with the <see cref="SyntaxAnnotation"/>.
         /// </summary>
-        public static async Task<Document> AddImportsAsync(Document document, SyntaxAnnotation annotation, OptionSet options = null, CancellationToken cancellationToken = default)
+        public static Task<Document> AddImportsAsync(Document document, SyntaxAnnotation annotation, OptionSet options = null, CancellationToken cancellationToken = default)
         {
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-            return await AddImportsAsync(document, root.GetAnnotatedNodesAndTokens(annotation).Select(t => t.FullSpan), options, cancellationToken).ConfigureAwait(false);
+            return AddImportsFromSyntaxesAsync(document, annotation, false, options, cancellationToken);
         }
 
         /// <summary>
         /// Adds namespace imports / using directives for namespace references found in the document within the spans specified.
         /// </summary>
         public static Task<Document> AddImportsAsync(Document document, IEnumerable<TextSpan> spans, OptionSet options = null, CancellationToken cancellationToken = default)
+        {
+            return AddImportsFromSyntaxesAsync(document, spans, false, options, cancellationToken);
```

Comment:
i'm actually curious if we want these to be unsafe.  but it's prob a reasonable thing for the PR for now to have this preserve compat.  we should have it be discussed if it's ok to actually just change the behavior to be always safe.

Reply:
See my comment on https://github.com/dotnet/roslyn/pull/37228#discussion_r304078733

Reply:
yeah, let's opt for compat for now :)